t/api/01-workers.t

```diff
@@ -147,5 +148,22 @@ $ret = $t->post_ok('/api/v1/workers', form => $worker_caps);
 is($ret->tx->res->code,       200, "register new worker");
 is($ret->tx->res->json->{id}, 3,   "new worker id is 3");
 
+subtest 'delete offline worker' => sub {
+    my $offline_worker_id = 9;
+    my $database_workers  = $t->app->schema->resultset("Workers");
+    $database_workers->create(
+        {
+            id        => $offline_worker_id,
+            host      => 'offline_test',
+            instance  => 5,
+            t_updated => time2str('%Y-%m-%d %H:%M:%S', time - 1200, 'UTC'),
+        });
+
+    $ret = $t->delete_ok("/api/v1/workers/$offline_worker_id")->status_is(200, "delete offline worker successfully.");
+
+    $ret = $t->delete_ok("/api/v1/workers/99")->status_is(404, "The offline worker not found.");
+
+    $ret = $t->delete_ok("/api/v1/workers/1")->status_is(400, "The worker status is not offline.");
```

Comment:
I realize I didn't mention it before... would you mind adding a check for the event itself here as well? I recently introduced a helper for this:

    is_deeply(                                                                                                                          
        OpenQA::Test::Case::find_most_recent_event($t->app->schema, 'worker_delete'),
        {id => "$offline_worker_id", name => "offline_test"},
        'Delete was logged correctly'
    );

Reply:
good suggestion, I got it. I will add a check for the audit event.